pilot/pkg/model/service.go

```diff
@@ -609,18 +610,45 @@ func ParseServiceKey(s string) (hostname Hostname, ports PortList, labels Labels
 	return
 }
 
+// defaultSubsetName - Use this name for subset is not defined.
+const defaultSubsetName = "_default_"
+
 // BuildSubsetKey generates a unique string referencing service instances for a given service name, a subset and a port.
 // The proxy queries Pilot with this key to obtain the list of instances in a subset.
 func BuildSubsetKey(direction TrafficDirection, subsetName string, hostname Hostname, port *Port) string {
-	return fmt.Sprintf("%s|%s|%s|%s", direction, port.Name, subsetName, hostname)
+	if subsetName == "" {
```

Comment:
Since this is such a core function, it would be good to update comment with behavior for missing subsetName, portName, maybe example output? Sorry to pile this on you, feel free to ignore and I'll take care of it and other cleanups.